daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1013,33 +1023,25 @@ object SBuiltin {
       }
```

Comment:
I wonder whether you want to check here that the contract ID in `args0` does not clash w.r.t. discriminators. If we reinterpret a tree, the contract ID might come from a `fetchByKey` construct which is translated into a fetch node only. So the fetched contract ID might come out of nowhere. But I'm not sure whether this can happen on this code path.

Reply:
Here I do not want to check the contract in `args0`. 
The check of contract gathering by contract key should be done in  `SBULookupKey` (done) and in `SBUFetchKey` (currently missing)